Adopt style(9) using clang-format

mchoo7 opened the pull request:
I find atk's coding style doesn't follow some conventions, such as having space in `<>` or putting pointer `*` or reference symbol `&` with variable types, not variable names.

FreeBSD's coding style style(9) can be applied here. One main exception in style(9) opposed to the "convention" is that style(9) use 8-space indentation while most projects use 4-space indentation. (2-space indentation is predominant in web dev). We can discuss whether or not to use 8-space indentation if needed.

ngie-eign reviewed (changes requested):
@fel1x-developer : for the sake of pragmatism, I think it's best to apply this change on a per-file basis since this is going to cause a lot of churn for little to no immediate benefit.
Also, there are a number of conflicts that need to be resolved in this change.

mchoo7 committed:
Adopt style(9) using clang-format

This commit adopts FreeBSD's style(9). This file is retrieved from
https://github.com/freebsd/freebsd-src/blob/main/.clang-format.
Contributers can apply this style on modified files in future changes.

Signed-off-by: Minsoo Choo <minsoochoo0122@proton.me>

ngie-eign commented:
Was this pulled from [FreeBSD](https://github.com/freebsd/freebsd-src/blob/c2a005a0a97d98472d81c5c9606e5d6e546dd40a/.clang-format)? If so, it might be better to figure out a way to keep this file in sync, somehow, instead of it drifting too far from upstream (freebsd-src).

mchoo7 commented:
> Was this pulled from [FreeBSD](https://github.com/freebsd/freebsd-src/blob/c2a005a0a97d98472d81c5c9606e5d6e546dd40a/.clang-format)? If so, it might be better to figure out a way to keep this file in sync, somehow, instead of it drifting too far from upstream (freebsd-src).

For now, automatically tracking updates in the freebse-src upstream is impossible. We can use [post-commit hook](https://stackoverflow.com/a/19200883/14383723) as an alternative.

By the way, the `.clang-format` file is not updated quite often. It had only [2 major changes](https://github.com/freebsd/freebsd-src/commits/main/.clang-format) in the last 3 years, so I think updating manually from the upstream would be fine.

ngie-eign commented:
Could you please add a comment to the file to note where it came from (FreeBSD; git revision <...>)? Thanks :).

mchoo7 commented:
@ngie-eign Sorry this PR was mixed with other unrelated changes I was working on. We only need 1 commit where we bring `.clang-format` from freebsd-src.

We might need a git hook so people can format files according to `.clang-format`.

ngie-eign reviewed (commented):
I looked at doing a similar change.
I think your proposal to introduce clang-format makes logical sense, but the implementation is a very high-impact/high churn change.
It's best to take the upstream FreeBSD config, and adjust it to match the existing style to reduce churn.
Once the core memory leak issues have been resolved and CI is enabled, I would be game to do widespread reformatting of the source in the repo.
